common4j/src/main/com/microsoft/identity/common/java/crypto/StorageEncryptionManager.java

```diff
@@ -175,6 +181,9 @@ public byte[] encrypt(final byte[] plaintext)
         } catch (final InvalidAlgorithmParameterException e) {
             errCode = INVALID_ALG_PARAMETER;
             exception = e;
+        } catch (final Throwable e) {
+            errCode = UNKNOWN_ERROR;
```

Comment:
Can we find a more meaningful error code?

Reply:
This one is unknown though... as in, we don't know what Android throw out.

Reply:
But we at least know this is failing in encryption layer right? Can we create a new error code? Something like `ENCRYPTION_FAILED`?

Reply:
ENCRYPTION_FAILED might not be accurate in some cases- since the exception we got is in the key generation.

I guess i could create UNKNOWN_CRYPTO_ERROR?

Reply:
The method in which this is thrown seems to be called `encrypt` so I suspected in the encryption layer.

Anyway, any thing more specific than UNKNOWN_ERROR would be good so if you want `UNKNOWN_CRYPTO_ERROR` then that's fine.